File: DemoMethods/Basic/Highlights.cs
using System;
using System.Linq;
using System.Text;
using System.Web.Http;
using DemoMethods.Entities;
using DemoMethods.Helpers;
using DemoMethods.Indexes;
using Raven.Client;

namespace DemoMethods.Basic
{
    public partial class BasicController : DemoApiController
    {
        [HttpGet]
        [Demo("HighLights", DemoOutputType.String, demoOrder: 120)]
        public object HighLights(string searchTerm = "UK USA")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                FieldHighlightings highlightings;

                var results = session
                    .Advanced
                    .DocumentQuery<Company, CompaniesAndAddresses>()
                    .Highlight("Address", 128, 1, out highlightings)
                    .Search("Address", searchTerm)
                    .ToList();

                var builder = new StringBuilder()
                    .AppendLine("<ul>");

                foreach (var fragments in results.Select(result => highlightings.GetFragments(result.Id)))
                {
                    builder.AppendLine(string.Format("<li>{0}</li>", fragments.First()));
                }

                var ul = builder
                    .AppendLine("</ul>")
                    .ToString();

                return ul;
            }
        }
    }
}

File: DemoMethods/src/main/java/net/ravendb/demo/basic/Highlights.java
package net.ravendb.demo.basic;

import net.ravendb.abstractions.basic.Reference;
import net.ravendb.client.FieldHighlightings;
import net.ravendb.client.IDocumentSession;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Company;
import net.ravendb.demo.indexes.CompaniesAndAddresses;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@Controller
public class Highlights {

    @RequestMapping("/Basic/HighLights")
    public String highLights(
            @RequestParam(value = "address", defaultValue = "UK USA") String address) {

        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {

            Reference<FieldHighlightings> highlightingsRef = new Reference<>();

            List<Company> results = session.advanced()
                    .documentQuery(Company.class, CompaniesAndAddresses.class)
                    .highlight("Address", 128, 1, highlightingsRef)
                    .search("Address", address)
                    .toList();

            StringBuilder builder = new StringBuilder();

            builder.append("<ul>");

            for (Company result : results) {
                String[] fragments = highlightingsRef.value.getFragments(result.getId());
                for (String fragment : fragments) {
                    builder.append(String.format("<li>%s</li>", fragment));
                }
            }

            String ul = builder.append("</ul>").toString();
            return ul;
        }
    }
}
